superset/csvform.py

```diff
@@ -0,0 +1,205 @@
+"""Contains the logic to create cohesive forms on the explore view"""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from flask_babel import lazy_gettext as _
+
+from flask_appbuilder.fieldwidgets import BS3TextFieldWidget
+from flask_appbuilder.forms import DynamicForm
+
+from flask_wtf.file import FileField, FileAllowed, FileRequired
+from wtforms import (
+    SelectField, IntegerField, StringField)
+from wtforms.validators import DataRequired, Optional, NumberRange
+
+from superset import app, db
+import superset.models.core as models
+import logging
+
+
+config = app.config
+
+class CsvToDatabaseForm(DynamicForm):
+    # These are the fields exposed by Pandas read_csv()
+
+    # TODO: Add a must be alphanumeric, not numeric validator.
+    name = StringField(_('Table Name'),
+                       description=_('Name of table to be created '
+                                     'from csv data.'),
+                       validators=[DataRequired()],
+                       widget=BS3TextFieldWidget())
+    csv_file = FileField(_('CSV File'),
+                         description=_('Select a CSV file to be '
+                                       'uploaded to a database.'),
+                         validators=[FileRequired(),
+                                     FileAllowed(['csv'],
+                                                 _('CSV Files '
+                                                   'Only!'))])
+    all_datasources = []
+    try:
+      all_datasources = db.session.query(models.Database.sqlalchemy_uri, models.Database.database_name).all()
+    except Exception as e:
+      logging.info("Exception retreiving dbs: {}".format(e))
```

Comment:
Because whenever I run it, it works but it breaks when the CI is running. The error it produces is: 
sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) no such table: dbs [SQL: 'SELECT dbs.sqlalchemy_uri AS dbs_sqlalchemy_uri, dbs.database_name AS dbs_database_name \nFROM dbs']

It needs the dbs table that is not existent on the travis instance but the table will exist whenever someone runs superset.